### ISSUE TITLE
I can't change the image on the login screen

### ISSUE THREAD

**Issue by oewersson** [gdm-settings/gdm-settings#223]
### What happened?

I installed the GDM Settings application in order to customize my lock screen.

I did all the correct installation as it says in the project's README.md, but I didn't try several times and in different ways, I was unsuccessful when trying to place an image, remaining in the OS default, I made a change to the name that I wanted to appear on the screen login and I was successful.

### Expected behavior

I was hoping to be able to change the login screen to the wallpaper I want.

### To Reproduce

Fix the bug of not being able to place the image.

### Application Version

GDM 42.0

### Format/Source

Flatpak

### Environment

- **OS**:  Ubuntu 22.04.3 LTS
- **GNOME**: 42.9

**Comment by sellwanger** [comment]
Same Problem here with Zorin 17 (Ubuntu 22.04 base).
Any Fix for this?

**Comment by realmazharhussain** [comment]
> ### To Reproduce
>
> Fix the bug of not being able to place the image.

This section ("To Reproduce") is for a detailed description of the step-by-step process to reproduce the issue you are having.

You might think it is obvious, but it is not. Please, provide more info on how it is not successful. Does it say "Failed to apply settings"? Or does it apply successfully but still not take effect? Does the app spit out any warning or error when you apply settings if you run it via the Terminal?

**Comment by Jibun-no-Kage** [comment]
Seems to work find in Bookworm 12.4/12.5.

**Comment by Cardistymo** [comment]
I still cant change background color/image on Linux Mint 21.3 with GNOME 42.9 and GDM Settings 4.4 installed. Are there any updates or is there a fix for this?

**Comment by sellwanger** [comment]
> I still cant change background color/image on Linux Mint 21.3 with GNOME 42.9 and GDM Settings 4.4 installed. Are there any updates or is there a fix for this?

Same (still) on Zorin 17 :/

**Comment by jcamposz** [comment]
@realmazharhussain  I have detected (in my personal computer with Fedora) that I couldn't change background because it's classifying the background type as default.  I have modified the code to show image selector in this case. Is it this behavior correct??? It's the only way that I found to change that image
![imatge](https://github.com/user-attachments/assets/1932d5c7-8f3d-4907-bf58-d52d2f016594)

**Comment by TollerNamen** [comment]
> > ### To Reproduce
> > Fix the bug of not being able to place the image.
> 
> This section ("To Reproduce") is for a detailed description of the step-by-step process to reproduce the issue you are having.
> 
> You might think it is obvious, but it is not. Please, provide more info on how it is not successful. Does it say "Failed to apply settings"? Or does it apply successfully but still not take effect? Does the app spit out any warning or error when you apply settings if you run it via the Terminal?

Well for my case, to address your question since noone has semmed to have done that so far, on
Pop!_OS 22.04 LTS
flatpak version
I have the same issue of the image not applying.
I was able to change the appearance like the font, or the themes shell, icons, cursor.
Background Image is also set no problem. "Applying" behaves like the other changes I did. It just goes with the usual password prompt and everything seems to work, but when you log out, every change applies I've tried so far - except the background being the boring gray default. When i log back in, and start the app again, the image is still shown as applied, even though no real change took effect.

**Comment by TollerNamen** [comment]
@realmazharhussain would you have a look at the above please? ^

**Comment by realmazharhussain** [comment]
@TollerNamen Does using any other tool (e.g. the deprecated [gdm-tools](https://github.com/realmazharhussain/gdm-tools.git)) to change the background work? Does changing the background to a color work?

**Comment by TollerNamen** [comment]
> @TollerNamen Does using any other tool (e.g. the deprecated [gdm-tools](https://github.com/realmazharhussain/gdm-tools.git)) to change the background work? Does changing the background to a color work?

@realmazharhussain I have tried both, so changing the color with the app does not apply as well, but with the client tool you suggested ([gdm-tools](https://github.com/realmazharhussain/gdm-tools.git)), it was successful! I hope this information helps.

**Comment by TollerNamen** [comment]
ALRIGHT
I just changed the Top Bar to black color, and color applying or image applying suddenly worked like a breeze!
Just noticed when i changed that setting in the app, but suddenly my background img which i set through the gdm-tools was gone and that color was applied still set in the app.

**Comment by TollerNamen** [comment]
![image](https://github.com/user-attachments/assets/7aa3fd19-ee7d-41a9-916f-390f08d5d75b)
and suddenly everything worked.
This was set to off when i had the problems.

**Comment by TollerNamen** [comment]
No, the issue is somehow non-existent anymore, background images are applied and appear no problem.
@realmazharhussain

**Comment by javabeans13** [comment]
Hey @realmazharhussain I don't know if this will help. But when I'm running gdm-settings on my sandbox laptop, without full disk encryption the tool works fine. When I'm using my daily driver, with full disk encryption, I cannot change any of the settings for login screen. I've also tried the old tools, but nothing seems to work.

Not sure if that will help, but on the ubuntu machine without disk encryption gdm-settings works fine, no issues.

**Comment by javabeans13** [comment]
> No, the issue is somehow non-existent anymore, background images are applied and appear no problem. [@realmazharhussain](https://github.com/realmazharhussain)

Unfortunately, it still exits. May I ask if you can share what other settings you have?

**Comment by migtorr** [comment]
If you still having this problem a workaround is copying your background to /usr/local/share

Try with `sudo cp /home/user/Pictures/yourimage /usr/local/share`

Then from GDM Settings select the image you copied at /usr/local/share

**Comment by Jibun-no-Kage** [comment]
I tend to use /usr/share/backgrounds (already exists in Debian 12) for all the wallpaper, backgrounds, etc. on Debian 12.  That seems to work too.  I do use png or jpg file types, the webp file type sometimes is not recognized?  GRUB same issue as well at times, BTW.

On Debian 12.10, seems to be working, in setting the background, enable logo, enable banner.  GNOME is only 43.9, so control of the highlight color is out of scope... I did force GNOME to 48 and then 48.1, and the highlight color seems to work in the GNOME shell, but not in GDM3 scope?  Maybe I missed something?  Which was a surprise, thinking or expecting the highlight color customization would work in GDM3?  Going to redo setup and try again... confirm if it is consistent behavior.

Have not looked at the code... but is the highlight color change being done by API call or change of the gresource file?  I just recently worked out how customize the css file for GDM3, extracting, changing, compiling the customized css file.  A few people have documented the process, but no one seems to have really explained the specifics of the css file content as applicable to GDM3 use, what changes what... step by step.  For example replaced all the 'blue' highlight color with 'purple' and when mouse hovers over element... the color is still blue?  So GDM3.css is not reference for everything?  Theme is always whatever the default is, never changed it.

**Comment by jastorino** [comment]
I am having this same issue on Zorin OS 18.

All other settings seem to work fine except for the background image.  I can't remove it or change it.
I am trying to use one of the backgrounds that comes with the system in /usr/share/backgrounds.
My disk is not encrypted.

Thanks for your help
